backend/services/application.py

```diff
@@ -0,0 +1,230 @@
+        """Returns application(s) for a specific user
+
+        Returns:
+            list[Application]: List of all current applications for a specific user
+        """
+
+        application_entity = (
+            self._session.query(ApplicationEntity)
+            .filter(ApplicationEntity.user_id == subject.id)
+            .first()
+        )
+
+        if application_entity is None:
+            return None
+
+        sections = (
+            self._session.query(section_application_table)
+            .filter(section_application_table.c.application_id == application_entity.id)
+            .order_by(section_application_table.c.preference)
+            .all()
+        )
+
+        section_ids = [section[1] for section in sections]
+
+        sections_entity = []
+        for id in section_ids:
+            sections_entity.append(
+                self._session.query(SectionEntity)
+                .filter(SectionEntity.id == id)
+                .first()
+            )
+
+        section_dict = {
+            i: section.to_model() for i, section in enumerate(sections_entity)
+        }
+
+        application = application_entity.map_application_to_detail_model(section_dict)
+
+        return UserApplication(application=application)
+    
+
+    def create_undergrad(self, application: New_UTADetails) -> New_UTADetails:
+        """
+        Creates an application based on the input object and adds it to the table.
+        If the application's ID is unique to the table, a new entry is added.
+        If the application's ID already exists in the table, it raises an error.
+
+        Parameters:
+            application (New_UTADetails): A new UTA application to add to table
+
+        Returns:
+            Application: Object added to table
+        """
+
+        if application.id:
+            application.id = None
+
+        application_entity = New_UTA_Entity.from_model(application)
+
+        application_entity.preferred_sections = (
+            self._session.query(SectionEntity)
+            .filter(
+                SectionEntity.id.in_(
+                    section.id for section in application.preferred_sections
+                )
+            )
+            .all()
+        )
+
+        self._session.add(application_entity)
+        self._session.commit()
+
+        for index, section in enumerate(application.preferred_sections):
+
+            self._session.execute(
+                update(section_application_table)
+                .filter(
+                    section_application_table.c.application_id == application_entity.id,
+                    section_application_table.c.section_id == section.id,
+                )
+                .values(preference=index)
+            )
+
+        self._session.commit()
+
+        return application_entity.to_details_model()
+
+    def update_undergrad(
+        self, subject: User, application: New_UTADetails
+    ) -> New_UTADetails:
```

Comment:
Same thing here, I think this update function could potentially wipe returning user data